specs/proposals.md

```diff
@@ -35,24 +37,62 @@ are part of later specification milestones.
 
-The proposer may also delete multiple output roots by calling the `deleteL2Outputs()` function and specifying the
+### L2OutputOracle v1.0.0
+
+The submission of output proposals is permissioned to a single account. It is expected that this
+account continues to submit output proposals over time to ensure that user withdrawals do not halt.
+
+The [L2 output proposer](../op-proposer) is expected to submit output roots on a deterministic
+interval based on the configured `SUBMISSION_INTERVAL` in the `L2OutputOracle`. The larger
+the `SUBMISSION_INTERVAL`, the less often L1 transactions need to be sent to the `L2OutputOracle`
+contract, but L2 users will need to wait a bit longer for an output root to be included in L1 (the settlement layer)
+that includes their intention to withdrawal from the system.
+
+The honest `op-proposer` algorithm assumes a connection to the `L2OutputOracle` contract to know
+the L2 block number that corresponds to the next output proposal that must be submitted. It also
+assumes a connection to an `op-node` to be able to query the `optimism_syncStatus` RPC endpoint.
+
+```python
+import time
+
+while True:
+    next_checkpoint_block = L2OutputOracle.nextBlockNumber()
+    rollup_status = op_node_client.sync_status()
+    if rollup_status.finalized_l2.number >= next_checkpoint_block:
+        output = op_node_client.output_at_block(next_checkpoint_block)
+        tx = send_transaction(output)
+    time.sleep(poll_interval)
+```
+
+A `CHALLENGER` account can delete multiple output roots by calling the `deleteL2Outputs()` function and specifying the
 index of the first output to delete, this will also delete all subsequent outputs.
 
-> **Note regarding future work:** In the initial version of the system, the proposer will be the same entity as the
-> sequencer, which is a trusted role. In the future proposers may need to submit a bond in order to post L2 output
-> roots, and some or all of this bond may be taken in the event of a faulty proposal.
+### L2OutputOracle v2.0.0
+
+The submission of output proposals is permissionless and there is no interval at which output
+proposals must be submitted at. It is expected that users will "just in time" propose an output
+proposal to facilitate their own withdrawal. A bond must be placed with an output proposal to
+disincentivize the proposal of malicious outputs. If it can be proven that the output is malicious,
+either via fault proof or by an attestation proof, then the bond can be slashed and used as a
+payment to the users who paid for gas to remove the malicious output.
+
+The `op-proposer` can still be used to submit output proposals. A naive implementation of the
+`op-proposer` will still submit output proposals on an interval. A more ideal implementation
+will use heuristics such as time of last submission or number of pending withdrawals that have
+yet to be included in an output proposal.
+
+Version `v2.0.0` includes breaking changes to the `L2OutputOracle` ABI.
```

Comment:
This section is pretty vague, but seems 'good enough' for now.

Reply:
You are right, will follow up with additions as we work on it